app/assets/javascripts/discourse/app/controllers/sidebar-section-form.js

```diff
@@ -0,0 +1,228 @@
+import Controller from "@ember/controller";
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+import { ajax } from "discourse/lib/ajax";
+import { isEmpty } from "@ember/utils";
+import { extractError } from "discourse/lib/ajax-error";
+import { inject as service } from "@ember/service";
+import I18n from "I18n";
+import { sanitize } from "discourse/lib/text";
+import { tracked } from "@glimmer/tracking";
+import { A } from "@ember/array";
+
+class Section {
+  @tracked title;
+  @tracked links;
+
+  constructor({ title, links, id }) {
+    this.title = title;
+    this.links = links;
+    this.id = id;
+  }
+
+  get valid() {
+    const validLinks =
+      this.links.length > 0 && this.links.every((link) => link.valid);
+    return this.validTitle && validLinks;
+  }
+
+  get validTitle() {
+    return !isEmpty(this.title);
+  }
+
+  get titleCssClass() {
+    return this.title === undefined || this.validTitle ? "" : "warning";
+  }
+}
+class SectionLink {
+  @tracked name;
+  @tracked value;
+
+  constructor({ router, name, value, id }) {
+    this.router = router;
+    this.name = name;
+    this.value = value ? `${this.protocolAndHost}${value}` : value;
+    this.id = id;
+  }
+  get protocolAndHost() {
+    return window.location.protocol + "//" + window.location.host;
+  }
+
+  get path() {
+    return this.value?.replace(this.protocolAndHost, "");
+  }
+
+  get valid() {
+    return this.validName && this.validValue;
+  }
+
+  get validName() {
+    return !isEmpty(this.name);
+  }
+
+  get nameCssClass() {
+    return this.name === undefined || this.validName ? "" : "warning";
+  }
+
+  get validValue() {
+    return (
+      !isEmpty(this.value) &&
+      (this.value.startsWith(this.protocolAndHost) ||
+        this.value.startsWith("/")) &&
+      this.path &&
+      this.router.recognize(this.path).name !== "unknown"
+    );
+  }
+
+  get valueCssClass() {
+    return this.value === undefined || this.validValue ? "" : "warning";
+  }
+}
+
+export default Controller.extend(ModalFunctionality, {
```

Comment:
Since this is a new controller, will we be able to write this using native js classes instead?